File: ios/LegalTalkIndia/Data/Model/Wallet.swift
import Foundation

struct WalletTransaction: Codable, Identifiable {
    let id: String
    let walletId: String
    let amount: Double
    let type: String // "deposit", "deduction", "credit", "withdrawal"
    let status: String // "pending", "completed", "failed"
    let description: String
    let timestamp: String
}

struct WalletResponse: Codable {
    let balance: Double
    let transactions: [WalletTransaction]
}

struct RegisterResponse: Codable {
    let user: User
    let profile: LawyerProfile?
}

struct AIAssistantRequest: Codable {
    let prompt: String
}

struct AIAssistantResponse: Codable {
    let text: String
}

File: android/app/src/main/java/com/legaltalk/india/data/model/Wallet.kt
package com.legaltalk.india.data.model

data class WalletTransaction(
    val id: String,
    val walletId: String,
    val amount: Double,
    val type: String, // "deposit", "deduction", "credit", "withdrawal"
    val status: String, // "pending", "completed", "failed"
    val description: String,
    val timestamp: String
)

data class WalletResponse(
    val balance: Double,
    val transactions: List<WalletTransaction>
)

data class WithdrawalRequest(
    val lawyerId: String,
    val amount: Double,
    val bankHolderName: String,
    val bankAccountNumber: String,
    val ifscCode: String
)

data class AIAssistantRequest(
    val prompt: String
)

data class AIAssistantResponse(
    val text: String
)

data class RegisterResponse(
    val user: User,
    val profile: LawyerProfile? = null
)

data class AdminMetrics(
    val totalLawyers: Int,
    val commissionRevenue: Double
)

data class AdminMetricsResponse(
    val metrics: AdminMetrics,
    val lawyerProfiles: List<LawyerProfile>
)
